cdap-app-fabric/src/main/java/co/cask/cdap/lineage/field/FieldLineageInfo.java

```diff
@@ -138,10 +161,120 @@ public long getChecksum() {
     return operations;
   }
 
+  public Map<EndPoint, Set<String>> getDestinationFields() {
+    if (destinationFields == null) {
+      destinationFields = computeDestinationFields();
+    }
+    return destinationFields;
+  }
+
+  public Map<EndPointField, Set<EndPointField>> getOutgoingSummary() {
+    if (outgoingSummary == null) {
+      outgoingSummary = computeOutgoingSummary();
+    }
+    return outgoingSummary;
+  }
+
+  public Map<EndPointField, Set<EndPointField>> getIncomingSummary() {
+    if (incomingSummary == null) {
+      incomingSummary = computeIncomingSummary();
+    }
+    return incomingSummary;
+  }
+
   private long computeChecksum() {
     return fingerprint64(canonicalize().getBytes(Charsets.UTF_8));
   }
 
+  private Map<EndPoint, Set<String>> computeDestinationFields() {
+    Map<EndPoint, Set<String>> destinationFields = new HashMap<>();
+    for (Operation operation : this.operations) {
```

Comment:
why not iterate only over the writeOperations?

Reply:
fixed.